sql/core/src/test/scala/org/apache/spark/sql/CacheHintsSuite.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.execution.CachedData
+import org.apache.spark.sql.execution.analysis.ResolveCacheHints.NO_RESULT_CACHE_TAG
+import org.apache.spark.sql.test.SharedSparkSession
+
+class CacheHintsSuite extends PlanTest with SharedSparkSession {
+  private def lookupCachedData(df: DataFrame): Option[CachedData] = {
+    spark.sharedState.cacheManager.lookupCachedData(df)
+  }
+
+  test("RESULT_CACHE Hint") {
+    val e = intercept[AnalysisException](
+      sql("SELECT /*+ RESULT_CACHE(abc) */ id from range(0, 1)").collect())
```

Comment:
is it the same as `CACHE TABLE abc AS SELECT id from range(0,1)`?

Reply:
They are very similar with few differences:

1. the result set is not the same
2. it skips caching step when already cached, so it will not fail


Parameters are disabled now because

1. I don't know if it is good to do parsing table identifiers in post-resolution 
2. oracle does not have params support either, maybe it's simple and easy for users？
3. minimize the patch size

Reply:
I don't think `CACHE TABLE` command fails if the plan is already cached. So basically the hint is to combine two sql statements?
```
CACHE TABLE abc AS SELECT id from range(0,1)
SELECT * FROM abc
```

Reply:
a quick test

```sql
 CACHE TABLE abc AS SELECT id from range(0,1);
Time taken: 0.673 seconds
spark-sql> CACHE TABLE abc AS SELECT id from range(0,1);
Error in query: Temporary view 'abc' already exists
```

And, yes, the RESULT_CACHE hint basically the same as the above 2 SQLs. Maybe with keyword LAZY is more precise.

Reply:
oh, i just notice another difference for RESULT_CACHE is to remove the NO_RESULT_CACHE_TAG